cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/draft/DraftStore.java

```diff
@@ -0,0 +1,300 @@
+    .withPrimaryKeys(NAMESPACE_COL, GENERATION_COL, OWNER_COL, ID_COL)
+    .build();
+  private static final Gson GSON = new Gson();
+  private final TransactionRunner transactionRunner;
+
+  public DraftStore(TransactionRunner transactionRunner) {
+    this.transactionRunner = transactionRunner;
+  }
+
+  /**
+   * @param namespace the namespace to fetch the drafts from
+   * @param owner the id of the owner of the drafts
+   * @param sortRequest The sorting that should be applied to the results, pass null if no sorting is required.
+   * @return a list of drafts
+   */
+  public List<Draft> listDrafts(NamespaceSummary namespace, String owner,
+                                SortRequest sortRequest) throws TransactionException {
+    AtomicReference<List<StructuredRow>> results = new AtomicReference<>();
+    List<Field<?>> prefix = new ArrayList<>(3);
+    prefix.add(Fields.stringField(NAMESPACE_COL, namespace.getName()));
+    prefix.add(Fields.longField(GENERATION_COL, namespace.getGeneration()));
+    prefix.add(Fields.stringField(OWNER_COL, owner));
+
+    transactionRunner.run(context -> {
+      StructuredTable table = context.getTable(TABLE_ID);
+      Range range = Range.singleton(prefix);
+      List<StructuredRow> temp = new ArrayList<>();
+      try (CloseableIterator<StructuredRow> rowIter = table.scan(range, Integer.MAX_VALUE)) {
+        while (rowIter.hasNext()) {
+          temp.add(rowIter.next());
+        }
+      }
+      results.set(temp);
+    });
+
+    List<StructuredRow> sortedResults = doSort(results.get(), sortRequest);
+    return sortedResults.stream().map(this::fromRow).collect(Collectors.toList());
+
+  }
+
+  /**
+   * Helper method to apply the sorting onto a list of rows. Sorting needs to take place at the store-level so it can
+   * leverage the StructuredRow to enable sorting on any field.
+   *
+   * @param rows list of {@link StructuredRow} to be sorted
+   * @param sortRequest {@link SortRequest} describing the sort to be performed
+   * @return a sorted list of {@link StructuredRow}
+   */
+  private List<StructuredRow> doSort(List<StructuredRow> rows, SortRequest sortRequest) {
+    if (sortRequest == null) {
+      return rows;
+    }
+    String sortField = sortRequest.getFieldName();
+    Map<String, FieldType.Type> fieldTypes = TABLE_SPEC
+      .getFieldTypes()
+      .stream()
+      .collect(Collectors.toMap(FieldType::getName, FieldType::getType));
+    if (!fieldTypes.containsKey(sortField)) {
+      throw new IllegalArgumentException(
+        String
+          .format("Invalid value '%s' for sortBy. This field does not exist in the Drafts table.",
+                  sortField));
+    }
+
+    rows.sort((o1, o2) -> {
+      int result = 0;
+      FieldType.Type fieldType = fieldTypes.get(sortField);
+
+      Object left = null;
+      Object right = null;
+      switch (fieldType) {
+        case STRING:
+          left = o1.getString(sortField);
```

Comment:
Instead of doing left right like this, you can based on the fieldType to use different comparator and delegate to it. Check out static method on the `Comparator`.

Reply:
done